Fix MacOS CI

Homebrew's CMake download location gives 404: {"errors":[{"code":"UNAUTHORIZED","message":"authentication required"}]}. Seen in e.g. #2483. Here is a copy of the log:
```
Starting: Build and test
==============================================================================
Task         : Bash
Description  : Run a Bash script on macOS, Linux, or Windows
Version      : 3.182.0
Author       : Microsoft Corporation
Help         : https://docs.microsoft.com/azure/devops/pipelines/tasks/utility/bash
==============================================================================
Generating script.
========================== Starting Command Output ===========================
/bin/bash --noprofile --norc /Users/runner/work/_temp/8d845d24-e341-4ad0-a9dd-d2397d5c9f47.sh
+ curl -O https://raw.githubusercontent.com/Homebrew/homebrew-core/de946dd0c0a91ecf6c78c843a4580bd7116c6bf0/Formula/cmake.rb
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100  1917  100  1917    0     0  50447      0 --:--:-- --:--:-- --:--:-- 50447
+ brew unlink cmake
Unlinking /usr/local/Cellar/cmake/3.20.1... 36 symlinks removed.
+ brew install ./cmake.rb
Warning: Calling `cellar` in a bottle block is deprecated! Use `brew style --fix` on the formula to update the style or use `sha256` with a `cellar:` argument instead.
Warning: Calling `sha256 "digest" => :tag` in a bottle block is deprecated! Use `brew style --fix` on the formula to update the style or use `sha256 tag: "digest"` instead.
Warning: Calling `sha256 "digest" => :tag` in a bottle block is deprecated! Use `brew style --fix` on the formula to update the style or use `sha256 tag: "digest"` instead.
Warning: Calling `sha256 "digest" => :tag` in a bottle block is deprecated! Use `brew style --fix` on the formula to update the style or use `sha256 tag: "digest"` instead.
Warning: Calling `sha256 "digest" => :tag` in a bottle block is deprecated! Use `brew style --fix` on the formula to update the style or use `sha256 tag: "digest"` instead.
Error: Failed to load cask: ./cmake.rb
Cask 'cmake' is unreadable: wrong constant name #<Class:0x00007ff101304ec8>
Warning: Treating ./cmake.rb as a formula.
==> Downloading https://ghcr.io/v2/homebrew/core/cmake/manifests/3.18.4
curl: (22) The requested URL returned error: 404 
Error: Failed to download resource "cmake_bottle_manifest"
Download failed: https://ghcr.io/v2/homebrew/core/cmake/manifests/3.18.4
+ c++ --version
Apple clang version 12.0.0 (clang-1200.0.32.29)
Target: x86_64-apple-darwin19.6.0
Thread model: posix
InstalledDir: /Applications/Xcode_12.4.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
+ cmake --version
/Users/runner/work/_temp/8d845d24-e341-4ad0-a9dd-d2397d5c9f47.sh: line 8: cmake: command not found
+ ctest -S /Users/runner/work/1/ITK-dashboard/dashboard.cmake -VV -j 4
/Users/runner/work/_temp/8d845d24-e341-4ad0-a9dd-d2397d5c9f47.sh: line 10: ctest: command not found
##[error]Bash exited with code '127'.
Finishing: Build and test
```

--- a/Testing/ContinuousIntegration/AzurePipelinesMacOS.yml
+++ b/Testing/ContinuousIntegration/AzurePipelinesMacOS.yml
@@ -60,10 +60,6 @@ jobs:
     - bash: |
         set -x
 
-        curl -O https://raw.githubusercontent.com/Homebrew/homebrew-core/de946dd0c0a91ecf6c78c843a4580bd7116c6bf0/Formula/cmake.rb
-        brew unlink cmake
-        brew install ./cmake.rb
-
         c++ --version
         cmake --version
 

--- a/Testing/ContinuousIntegration/AzurePipelinesMacOSPython.yml
+++ b/Testing/ContinuousIntegration/AzurePipelinesMacOSPython.yml
@@ -68,10 +68,6 @@ jobs:
     - bash: |
         set -x
 
-        curl -O https://raw.githubusercontent.com/Homebrew/homebrew-core/de946dd0c0a91ecf6c78c843a4580bd7116c6bf0/Formula/cmake.rb
-        brew unlink cmake
-        brew install ./cmake.rb
-
         c++ --version
         cmake --version
 

